install.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
+set -e
+
+temp_dir=$(mktemp -d /tmp/xh.XXXXXXXX)
+trap 'rm -r "$temp_dir"' EXIT INT TERM
+cd "$temp_dir"
+
+if [ "$(uname -s)" = "Darwin" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-apple-darwin"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-unknown-linux-musl"
+elif [ "$(uname -s)" = "Linux" ] && ( uname -m | grep -q -e '^arm' -e '^aarch' ); then
+    target="arm-unknown-linux-gnueabihf"
+else
+    echo "Unsupported OS or architecture"
+    exit 1
+fi
+
+if which curl > /dev/null; then
+    fetch='curl -sSL -o'
+elif which wget > /dev/null; then
+    fetch='wget -nv -O'
+else
+    echo "Can't find curl or wget, can't download package"
+    exit 1
+fi
+
+echo "Detected target: $target"
+
+url=$(
+    $fetch - https://api.github.com/repos/ducaale/xh/releases/latest |
+    tac | tac | grep -wo -m1 "https://.*$target.tar.gz" || true
+)
+if ! test "$url"; then
+    echo "Could not find release info"
+    exit 1
+fi
+
+echo "Downloading xh..."
+
+if ! $fetch xh.tar.gz "$url"; then
+    echo "Could not download tarball"
+    exit 1
+fi
+
+user_bin="$HOME/.local/bin"
+case $PATH in
+    *:"$user_bin":* | "$user_bin":* | *:"$user_bin")
+        default_bin=$user_bin
+        ;;
+    *)
+        default_bin='/usr/local/bin'
+        ;;
+esac
+
+printf "Install location [default: %s]: " "$default_bin"
+read -r bindir < /dev/tty
+bindir=${bindir:-$default_bin}
+
+while ! test -d "$bindir"; do
+    echo "Directory $bindir does not exist"
+    printf "Install location [default: %s]: " "$default_bin"
+    read -r bindir < /dev/tty
+    bindir=${bindir:-$default_bin}
+done
+
+tar xzf xh.tar.gz
+
+if test -w "$bindir"; then
+    mv xh-*/xh "$bindir/"
+    ln -sf "$bindir/xh" "$bindir/xhs"
```

Comment:
I feel like we should tell the user that we're doing this, but I can't think of a way that isn't noisy.

Reply:
Would listing each install location in its own line reduce the clutter?
```
Detected target: x86_64-unknown-linux-musl
Downloading xh...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   650  100   650    0     0   9420      0 --:--:-- --:--:-- --:--:--  9420
100 3357k  100 3357k    0     0  5381k      0 --:--:-- --:--:-- --:--:-- 8491k
Install location [default: /home/ducaale/.local/bin]:
xh 0.9.2 has been installed to:
 • /home/ducaale/.local/bin/xh
 • /home/ducaale/.local/bin/xhs
```

Reply:
Good idea, I like the way it looks.